cranelift-codegen/src/isa/x86/abi.rs

```diff
@@ -931,10 +915,17 @@ fn insert_common_epilogue(
         assert_eq!(cfa_state.current_depth, -word_size);
         assert_eq!(cfa_state.cf_ptr_offset, word_size);
 
-        let new_cfa = FrameLayoutChange::CallFrameAddressAt {
-            reg: cfa_state.cf_ptr_reg,
-            offset: cfa_state.cf_ptr_offset,
-        };
+        // Inserting preserve CFA state operation before FP pop instructions.
+        let mut new_cfa = vec![
+            FrameLayoutChange::Preserve,
+            FrameLayoutChange::CallFrameAddressAt {
+                reg: cfa_state.cf_ptr_reg,
+                offset: cfa_state.cf_ptr_offset,
+            },
+        ];
+        if is_last {
+            new_cfa.remove(0);
```

Comment:
I have a loose preference for making the right vec in the first place rather than conditionally removing an element especially since `is_last` is an argument now. But, lacking a benchmark I'm not sure if it really matters in release builds.